.github/workflows/build.yml

```diff
@@ -55,6 +55,9 @@ jobs:
       - name: Setup Rust cache
         uses: ./.github/workflows/setup-rust-cache
 
+      - name: Update Rust
+        run: rustup update
```

Comment:
I'm skeptical of such unpinned updates, but then again, I realize that the build machines on GH also using a floating version — probably tracking stable with some lag?

Perhaps it's okay since the Rust tooling is generally pretty conservative about adding breaking changes.

Reply:
Ah, I just spotted a small problem with this: we need to update Rust _before_ setting up the cache (just above. The cache is keyed on the Rust version active at the time the action is loaded.

Let me crate a PR for that.

Reply:
@qwandor, do you know how this update plays together with the tools already installed?

I see

```
  stable-x86_64-unknown-linux-gnu updated - rustc 1.82.0 (f6e511eec 2024-10-15) (from rustc 1.81.0 (eeb90cda1 2024-09-04))
warning: tool `rust-analyzer` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
warning: tool `rustfmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
warning: tool `cargo-fmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
```

in the latest run on `main`. It would be a shame if we now end up with `rustfmt` out of sync with `rustc`.